lib/cocoapods-core/podfile/target_definition.rb

```diff
@@ -174,7 +174,7 @@ def abstract=(abstract)
       # @return [String] the inheritance mode for this target definition.
       #
       def inheritance
-        get_hash_value('inheritance', 'complete')
+        get_hash_value('inheritance') || 'complete'
```

Comment:
why these changes?

Reply:
I've changed several getters to avoid mutating the hash because iterating over deps mutated them. 

The code assumed, previously, (and implicitly) that deps only get evaluated after the `Podfile` is completely scanned.

However, going over previously added deps before the entire `Podfile` is scanned mutated things and now it doesn't.

Reply:
:+1:. Is there a test that will fail if we accidentally go back to the old behavior later?

Reply:
want to continue this @igor-makarov ?

Reply:
Hadn't had the time to think about this yet.. Sorry 😭